Tests/TuistGeneratorTests/Generator/ProjectGeneratorTests.swift

```diff
@@ -48,6 +48,70 @@ final class ProjectGeneratorTests: TuistUnitTestCase {
         XCTAssertTrue(FileHandler.shared.exists(got.path.appending(RelativePath("../Derived/InfoPlists/Target.plist"))))
     }
 
+    func test_generate_doesNotWipeUserData() throws {
+        // Given
+        let temporaryPath = try self.temporaryPath()
+        let paths = try createFiles([
+            "Foo.xcodeproj/xcuserdata/a",
+            "Foo.xcodeproj/xcuserdata/b/c",
+        ])
+
+        let target = Target.test(name: "Target", platform: .iOS, product: .framework)
+        let project = Project.test(path: temporaryPath, name: "Foo", targets: [target])
+
+        let graph = Graph.create(project: project,
+                                 dependencies: [
+                                     (target: target, dependencies: []),
+                                 ])
+
+        // When
+        try (0 ..< 2).forEach { _ in
+            _ = try subject.generate(project: project, graph: graph)
```

Comment:
What about moving theses tests to `ProjectGeneratorIntegrationTests`. Based on the implementation it seems that we are not stubbing the calls to the subject's dependencies.

Reply:
makes sense, I believe the entire file can be moved there as the majority of tests here appear to be using the default implementations.